ui/src/pages/QuizPage.vue

```diff
@@ -0,0 +1,424 @@
+<template>
+  <div class="quizapp">
+    <!-- <QuizScreen /> -->
+    <!-- <nav class="navbar navbar-expand-lg navbar-dark bg-main-green">
+      <a class="navbar-brand">
+        <img
+          src="../assets/NFlaquizlogo.png"
+          width="30"
+          height="30"
+          class="d-inline-block align-top mr-2"
+          alt=""
+        />
+        NFlaquiz
+      </a>
+    </nav> -->
+    <div class="container">
+      <div class="row my-3">
+        <div class="col-12">
+          <h1 class="text-center main-title">国旗クイズ</h1>
+          <p class="text-center">国名を当てよう！</p>
+        </div>
+      </div>
+
+      <div class="row">
+        <!-- 問題数を表示する領域 -->
+        <div class="col-md-8 offset-md-2">
+          第{{ currentQuizNumber }}問 / 全{{ totalQuizNumber }}問
+          <button
+            class="hint-btn" 
+            data-toggle="modal" 
+            data-target="#hintModal"
+          >
+            ヒント
+          </button>
+        </div>
+
+        <div class="col-md-8 offset-md-2">
+          <div class="questionBox">
+            <span class="box-title"></span>
+            <p>この国旗の国名は？</p>
+            <img :src="flagImgPath" class="flagImg" />
+          </div>
+        </div>
+
+        <!-- 回答ボタンを表示する領域 -->
+        <div class="col-md-8 offset-md-2 text-center mt-5">
+          <div class="btn-box">
+            <button
+              v-for="(option, index) in options"
+              :key="index"
+              class="answer-btn answer-btn-bg1 mx-4"
+              data-toggle="modal"
+              data-target="#answerCheckModal"
+              @click="judgeAnswer(option)"
+            >
+              {{ option }}
+            </button>
+          </div>
+
+          <p class="text-center">正解の国名をクリックしよう！</p>
+        </div>
+      </div>
+    </div>
+
+    <!-- ヒントを表示するモーダル -->
+
+    <div 
+      class="modal"
+      id="hintModal"
+      role="dialog"
+      aria-labelledby="exampleModalCenterTitle"
+      aria-hidden="true"
+    >
+      <div class="modal-dialog modal-dialog-centered" role="document">
+        <div class="modal-content">
+          <div class="modal-header">
+            ヒント
+            <button class="hint-close" data-dismiss="modal">×</button>
+          </div>
+
+          <div class="modal-body" v-for="(hint, number) in hints" :key="hint">
+            ヒント{{ number+1 }}：{{ hint }}
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <!-- 答えを表示するモーダル -->
+
+    <div
+      class="modal"
+      id="answerCheckModal"
+      tabindex="-1"
+      role="dialog"
+      aria-labelledby="exampleModalCenterTitle"
+      aria-hidden="true"
+    >
```

Comment:
答えのモーダル外を押すとモーダルが閉じてしまい，何回でも解答できてしまうため，
モーダル外を押しても閉じないようにしたほうがいいかもしれない

```suggestion
      data-backdrop="static"
    >
```

Reply:
修正しました！